packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -1911,18 +1911,21 @@ export class Task {
     private updateDefinitionBasedOnExecution(): void {
         if (this.taskExecution instanceof ProcessExecution) {
             Object.assign(this.taskDefinition, {
+                type: 'process',
```

Comment:
I don't think we should ever update the `type` field: in Eclipse Che, we register a `TaskRunner`for the task type 'che'. Once the `type` field is overwritten, we can't get it back.

Reply:
This update has been there for two years before the introduction of customExecution,
I am trying to understand how this has not caused problems for 'che' then ?

Reply:
@alvsan09 It's not a problem if you don't have an execution set in the task (which happens to be the case in Che), but otherwise, you'll never match the task type. So while the problem might not immediately manifest, it's still there.
Generally, I think https://github.com/eclipse-theia/theia/pull/9377 offers a cleaner way forward.

Reply:
@tsmaeder, this PR would be a reset to base functionality in case a more complete solution is not available in time,
I will test and review #9377